x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rest/BaseInferenceAction.java

```diff
@@ -60,4 +66,21 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
     protected abstract boolean shouldStream();
 
     protected abstract ActionListener<InferenceAction.Response> listener(RestChannel channel);
+
+    private String extractProductUseCase(RestRequest restRequest) {
+        var headers = restRequest.getHeaders();
+
+        if (Objects.isNull(headers) || headers.isEmpty()) {
+            return "";
+        }
+
+        var productUseCaseHeaders = headers.get(InferencePlugin.X_ELASTIC_PRODUCT_USE_CASE_HTTP_HEADER);
+
+        if (Objects.isNull(productUseCaseHeaders) || productUseCaseHeaders.isEmpty()) {
+            return "";
+        }
+
+        // We always get the first value as the header doesn't allow multiple values
+        return productUseCaseHeaders.getFirst();
```

Comment:
If we do backport this, it's going to complain about `getFirst` not being a call in 8.19. Might be worth just leave it as `get(0)` to avoid all that lol (I've run into it many times from experience).

Reply:
Good catch 🎣 I've also ran into this in the past, adjusted with [Use .get(0) instead of getFirst to avoid compilation errors in backport.](https://github.com/elastic/elasticsearch/pull/124025/commits/2f881e0fd286f99bc4a6a59a895afb1b09c121fb)